File: Assignment_7/Assignment7_1.py
"""-----------------------------------------------------------------------------
                          Assignment7_1 
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
------------------------------------------------------------------------------------------
Problem statement : Write a program using lambda function
                    -Lambda function to calculate square
                    -Lambda function to calculate cube of a number
----------------------------------------------------------------------------------------"""
#-------------------------------------------------------------------------------
# lambda functions to find square and cube of a number
#-------------------------------------------------------------------------------
square_number = lambda num : num * num
cube_number = lambda num : num**3
#-------------------------------------------------------------------------------
# main() function calls lambda functions for square and cube
#-------------------------------------------------------------------------------
def main():
    print("------------------------------------------------------")
    number=int(input("Enter a number to calculate sqaure and cube:"))
    print(f"\nSquare of {number} using lambda function is :",square_number(number))
    print(f"\nCube of {number} using lambda function is :",cube_number(number))
    print("------------------------------------------------------")

#-------------------------------------------------------------------------------
# Entry point of program
#-------------------------------------------------------------------------------
if __name__=="__main__":   
    main()
#-------------------------------------------------------------------------------

File: JavaPractice/Assignment_7/Assignment7_1.java
import java.util.Scanner;
import java.util.function.Consumer;

/*-----------------------------------------------------------------------------
                          Assignment7_1 
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
------------------------------------------------------------------------------------------
Problem statement : Write a program using lambda function
                    -Lambda function to calculate square
                    -Lambda function to calculate cube of a number
----------------------------------------------------------------------------------------*/
interface Arithmetic{

	public int processNum(int a);
}
public class Assignment7_1 {

	public static void main(String[] args) {
		Scanner inputObj = new Scanner(System.in);
		System.out.print("Enter number:"); 
		int number=inputObj.nextInt();
		Arithmetic square = (num)->num*num;
		System.out.println("Square is:"+square.processNum(number));
		
		
		Arithmetic cube = (num)->num*num*num;
		System.out.println("Cube is:"+cube.processNum(number));
		
		inputObj.close();
		
	}

}
